py/dml/ctree.py

```diff
@@ -541,6 +541,12 @@ def mkExpressionStatement(site, expr):
         return mkNull(site)
     return ExpressionStatement(site, expr)
 
+def toc_constsafe_pointer_assignment(site, source, target, typ):
```

Comment:
better naming and purity if you let the caller call `.toc()`

Reply:
err... Not sure, when I wrote this I did not mean for this to be used outside of C emit phase. True, it _could_ with its current implementation, but do I want it to _need_ to be able to be used in the codegen phase? I.e. do I want to restrict it from being able to call `out()`?

In other words, I dislike the fact that it would be expressed in terms of codegen expression/statements. If we care about purity, I'd much rather we express it in terms of a datatype representing C emitting.